src/app/Marine2/components/boxes/GeneratorRelays/GeneratorRelay.tsx

```diff
@@ -0,0 +1,118 @@
+import { observer } from "mobx-react-lite"
+import GeneratorIcon from "../../../images/icons/generator.svg"
+import { translate } from "react-i18nify"
+import { formatPower } from "../../../utils/format"
+import DeviceCompact from "../DeviceCompact"
+import { useActiveInValues } from "@elninotech/mfd-modules"
+import ValueBox from "../../ui/ValueBox"
+import AutoStartStopSetter from "../../ui/AutoStartStopSetter/AutoStartStopSetter"
+import { RELAY_FUNCTION } from "../../../utils/constants"
+
+const GeneratorRelay = ({
+  statusCode,
+  active,
+  phases = 1,
+  manualStart,
+  autoStart,
+  relayFunction,
+  updateManualMode,
+  updateAutoMode,
+  mode = "compact",
+}: Props) => {
+  const getGeneratorState = (statusCode: number, active: boolean, phases: number) => {
+    if (active) {
+      return phases > 1 ? translate("common.nrOfPhases", { phases }) : translate("common.running")
+    }
+
+    switch (statusCode) {
+      case 1:
+        return translate("common.running")
+      case 10:
+        return translate("common.error")
+      default:
+        return translate("common.stopped")
+    }
+  }
+
+  const title = translate("widgets.generator")
+  const subTitle = getGeneratorState(statusCode, active ?? false, phases)
+
+  const { current, voltage, power } = useActiveInValues()
+  const powerSum = active
+    ? power.reduce((sum: number, b) => {
+        return b ? sum + b : sum
+      }, 0)
+    : 0
+  const powerFormatted = formatPower(powerSum)
+  const unit = powerSum > 1000 ? "kW" : "W"
+
+  const phasesOverview = []
+  for (let phase = 0; phase < phases; phase++) {
+    phasesOverview.push([
+      { value: voltage[phase], unit: "V" },
+      { value: current[phase], unit: "A" },
+      { value: power[phase], unit: "W" },
+    ])
+  }
+
+  if (mode === "compact") {
+    return (
+      <DeviceCompact
+        icon={
+          <GeneratorIcon
+            /* todo: fix types for svg */
+            /* @ts-ignore */
+            className={"w-7"}
+          ></GeneratorIcon>
+        }
+        title={title}
+        subTitle={subTitle}
+        value={powerFormatted}
+        unit={unit}
+      />
+    )
+  }
+
+  return (
+    <ValueBox
+      icon={
+        <GeneratorIcon
+          /* todo: fix types for svg */
+          /* @ts-ignore */
+          className={"w-7"}
+        ></GeneratorIcon>
+      }
+      title={title}
+      buttons={
+        relayFunction === RELAY_FUNCTION.GENERATOR_START_STOP && statusCode !== undefined ? (
+          <AutoStartStopSetter
+            autoStart={autoStart}
+            isAutoStartDisabled={false}
+            updateAutoMode={updateAutoMode}
+            updateManualMode={updateManualMode}
+            manualStart={manualStart}
+          />
+        ) : undefined
+      }
+      unit={"A"}
+      value={current[0]} //TODO: get update to design for what should be shown when there are multiple phases
```

Comment:
I think we show total power in either W or kW in that case in other places right? Like AC loads